src/CMakeLists.txt

```diff
@@ -96,6 +96,38 @@ if(MSVC)
     target_compile_options(inexor-vulkan-renderer PRIVATE "-EHs")
 endif()
 
+# The marked constants in application.hpp will be replaced with the following values
+set(ENGINE_NAME "Inexor Engine")
+set(APP_NAME "Inexor Vulkan-renderer example")
+
+set(ENGINE_VERSION_MAJOR 0)
+set(ENGINE_VERSION_MINOR 1)
+set(ENGINE_VERSION_PATCH 0)
+
+set(APP_VERSION_MAJOR 1)
+set(APP_VERSION_MINOR 0)
+set(APP_VERSION_PATCH 0)
+
+set(BUILD_TYPE ${CMAKE_BUILD_TYPE})
+
+# Extract the current git sha
+find_package(Git REQUIRED)
+set(GIT_SHA "unknown")
+
+execute_process(
+    COMMAND ${GIT_EXECUTABLE} describe --always --abbrev=7
+    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+    ERROR_QUIET
+    OUTPUT_VARIABLE GIT_SHA
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+
+
+# Replace variables in header file with CMake values
+configure_file(
+    ${PROJECT_SOURCE_DIR}/include/inexor/vulkan-renderer/application.hpp.in ${PROJECT_SOURCE_DIR}/include/inexor/vulkan-renderer/application.hpp
+)
```

Comment:
Here you would wanna do something like
```cmake
configure_file(../include/inexor/vulkan-renderer/config.hpp.in include/inexor/vulkan-renderer/config.hpp)
```
and then add
```cmake
target_include_directories(inexor-vulkan-renderer PUBLIC ${CMAKE_CURRENT_BINARY_DIR}/include)
```
so that we can do `#include "inexor/vulkan-renderer/config.hpp"`. Not sure if it would work exactly like that mind, since doing this in `src/CMakeLists.txt` is a little akward (I normally do everything from root and then use `target_sources` in `src`).

Reply:
i like to call it `meta.hpp`

Reply:
Sure, it doesn't really matter so long as cmake isn't generating stuff in source dir.